src/Autosuggest.js

```diff
@@ -222,8 +238,9 @@ class Autosuggest extends Component {
               }
             } else if (suggestions.length > 0) {
               const { newFocusedSectionIndex, newFocusedItemIndex } = data;
+              const safeValue = valueBeforeUpDown ? valueBeforeUpDown : value;
               const newValue = newFocusedItemIndex === null ?
-                valueBeforeUpDown :
+                safeValue :
```

Comment:
Sure thing. 

The `valueBeforeUpDown` variable isn't always set and can be null, which raises an exception later in the chain . 

This is because when we call `updateFocusedSuggestion` in the mouse events we don't pass a `value` (and can't since there's no reference to it). So if you mouse over an item before up/down, `valueBeforeUpDown` stays null. If you then transition directly to the input field using up/down, then `valueBeforeUpDown` is still null but `value` is equal to the input field.

To reproduce:
- type 'c'
- mouse over the first suggestion
- press Up
- you will get a console error
- this also applies to the last suggestion and pressing down

This check ensures that if there's no `valueBeforeUpDown`, then `value`is used (which can only happen when you transition into the input field, and thus `value` is the value of the input field).

Here's a GIF of the bug (note: this is live on master, so this is unrelated to my changes):

![](http://dp.hanlon.io/2a023S2A2J2u/Screen%20Recording%202016-05-08%20at%2001.56%20PM.gif)